python/tank/hook.py

```diff
@@ -215,72 +242,45 @@ def get_publish_path(self, sg_publish_data):
         """
         Returns the path on disk for a publish entity in Shotgun.
 
-        Use this method if you have a shotgun publish entity and want
-        to get a local path on disk. This method will ensure that however
-        the publish path is encoded, a local path is returned.
+        Convenience method that calls :meth:`sgtk.util.resolve_publish_path`.
+
+        :param sg_publish_data: Dictionaries containing Shotgun publish data.
+            Each dictionary needs to at least contain a code, type, id and a path key.
 
-        :param sg_publish_data: Shotgun dictionary containing
-                                information about a publish. Needs to at least
-                                contain a type, id and a path key.
-        :returns: String representing a local path on disk.
+        :returns: A path on disk to existing file or file sequence.
+
+        :raises: :class:`~sgtk.util.PublishPathNotDefinedError` if the path isn't defined.
+        :raises: :class:`~sgtk.util.PublishPathNotSupported` if the path cannot be resolved.
         """
-        return self.get_publish_paths([ sg_publish_data ])[0]
+        # avoid cyclic refs
+        from .util import resolve_publish_path
+        return resolve_publish_path(self.sgtk, sg_publish_data)
 
     def get_publish_paths(self, sg_publish_data_list):
         """
         Returns several local paths on disk given a
         list of shotgun data dictionaries representing publishes.
 
-        Use this method if you have several shotgun publish entities and want
-        to get a local path on disk. This method will ensure that however
-        the publish path is encoded, a local path is returned.
+        Convenience method that calls :meth:`sgtk.util.resolve_publish_path`.
+
+        .. deprecated:: 0.18.64
```

Comment:
it was added as a premature optimization assuming we would need to resolve a number of publishes at once. Then never used. But it makes it hard to raise exceptions in a nice way and i am not even sure it's a good way to handle the case where you want higher performance across several operations.